lib/api.js

```diff
@@ -153,6 +216,11 @@ PlexAPI.prototype._authenticateAndRetry = function _authenticateAndRetry(url, re
         return Q.reject(new Error('Permission denied even after attempted authentication :( Wrong username and/or password maybe?'));
     }
 
+    // Bit of a kludgey way to generate an Identifier before we send off the options to auth.
+    // Right now it doesn't appear that using the same identifier on every request is necessary, but
+    // it's probably good practice to do so.
+    self.getIdentifier();
```

Comment:
As this ensures an identifier is always generated, should it be moved into the constructor instead? Something like

```js
this.options.identifier = this.options.identifier || uuid.v4();
```
on line #21?